syntax = "proto3";

service PetShop {
    // Supplies one kitten to the caller.
    rpc GetKitten(GetKittenRequest) returns (GetKittenResponse);
}

enum Breed {
    HALP = 0;
    PERSIAN = 1;
    BENGAL = 2;
    RUSSIAN_BLUE = 3;
    CORNISH_REX = 4;
}

message Kitten {
    string name = 1;
    Breed breed = 2;
    float weight = 3;
    int32 age = 4;
    int64 dob = 5;
    bool neutered = 6;
}

message GetKittenRequest {
    // Breed of kitten requested. If not given then the server
    // may provide a kitten of any breed.
    Breed breed = 1;
}

message GetKittenResponse {
    Kitten kitten = 1;
}
